### ISSUE TITLE
import/export: false positives for typescript overloads

### ISSUE THREAD

**Issue by Philipp91** [benmosher/eslint-plugin-import#1357]
Minimal example to reproduce (not a sensible use case for overloads, but anyway):

```
export function consume(val: number): void;
export function consume(val: string): void;
export function consume(val: number | string): void {
    // Do something.
}
```

All three `consume` will be annotated with ` error  Multiple exports of name 'consume'   import/export`.

**Comment by ljharb** [comment]
Actually do you need all the export keywords? I’d expect the first two to not need it.

**Comment by Philipp91** [comment]
Sorry, I should have mentioned this. Leaving them off results in a TypeScript compiler error: `TS2383: Overload signatures must all be exported or non-exported.`

**Comment by leethree** [comment]
I'm encountering the same issue.

**Comment by matt-tingen** [comment]
Here's a workaround:

```ts
interface Consume {
  (val: number): void;
  (val: string): void;
}
export const consume: Consume = (val: number | string) => {
  // ...
}
```

**Comment by jose-nunez** [comment]
The interface workaround works for me and looks nicer than exporting all the overloads.
Thanks!

**Comment by tylerthehaas** [comment]
The work around is good and nice that there is an alternate syntax but the plugin should be able to handle all the features of typescript

**Comment by ljharb** [comment]
Agreed; a PR to fix it is welcome.
